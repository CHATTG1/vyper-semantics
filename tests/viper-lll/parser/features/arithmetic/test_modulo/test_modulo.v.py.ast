%pgm(, ,
  %fdecl(%@public, num_modulo_num, , %num,
    %return(%binop(%, 1, 2)))
  %fdecl(%@public, decimal_modulo_decimal, , %decimal,
    %return(%binop(%, %fixed10(15, 10), %fixed10(33, 100))))
  %fdecl(%@public, decimal_modulo_num, , %decimal,
    %return(%binop(%, %fixed10(5, 10), 1)))
  %fdecl(%@public, num_modulo_decimal, , %decimal,
    %return(%binop(%, %fixed10(15, 10), 1)))
)
