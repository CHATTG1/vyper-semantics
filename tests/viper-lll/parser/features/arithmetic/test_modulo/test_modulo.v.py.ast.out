$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 3369946726 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $clamp ( $mload ( 96 ) , $smod ( 1 , $clamp_nonzero ( 2 ) ) , $mload ( 64 ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 1753020054 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $clamp ( $mload ( 160 ) , $smod ( 15000000000 , $clamp_nonzero ( 3300000000 ) ) , $mload ( 128 ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 973439414 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $clamp ( $mload ( 160 ) , $smod ( 5000000000 , $mul ( $clamp_nonzero ( 1 ) , 10000000000 ) ) , $mload ( 128 ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 3886547014 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $clamp ( $mload ( 160 ) , $smod ( 15000000000 , $mul ( $clamp_nonzero ( 1 ) , 10000000000 ) ) , $mload ( 128 ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
